platform/linuxbsd/x11/display_server_x11.cpp

```diff
@@ -2998,7 +2998,11 @@ bool DisplayServerX11::window_is_focused(WindowID p_window) const {
 
 	const WindowData &wd = windows[p_window];
 
-	return wd.focused;
+	Window focused_window;
+	int focus_ret_state;
+	XGetInputFocus(x11_display, &focused_window, &focus_ret_state);
+
+	return wd.x11_window == focused_window;
```

Comment:
Should we update `wd.focused` accordingly after doing this synchronous check?

Reply:
My personal inclination would be not to update `wd.focused` here.

There's a bunch of other state that changes when the `FocusIn` and `FocusOut` events are received, and unless we're going to also run all of that, I don't want to just change `wd.focused`, which could lead to the internal state of `DisplayServerX11` being temporarily inconsistent (temporary, because everything else will get change when the event finally comes in).

But if other folks who are more expert in this code (ex @bruvzg) think it would be better to update `wd.focused` here, I'd be happy to defer to their judgement and update the PR :-)